src/server/storage-controller.js

```diff
@@ -206,6 +220,19 @@ class StorageController {
       validateParams(deleteStoragePlanParamsSchema),
       (req, res) => this.handleDeleteStoragePlan(req, res)
     );
+
+    router.get(
+      '/api/v1/storage/room-media-overview',
+      needsPermission(permissions.BROWSE_STORAGE),
+      (req, res) => this.handleGetRoomMediaOverview(req, res)
+    );
+
+    router.delete(
+      '/api/v1/storage/room-media/:roomId/:name',
+      needsPermission(permissions.DELETE_OWN_PRIVATE_CONTENT),
+      validateParams(deleteRoomMediaParamsSchema),
+      (req, res) => this.handleDeleteRoomMedia(req, res)
+    );
```

Comment:
By introducing this set of simplified CRUD operations on room media, we can get rid of:
* the more complicated operations based on path analysis, we can just fetch, upload and delete by roomId + file name
* the two middlewares in `storage-controller` that inject the storage and storage plan into each request, as these endpoints always return all information needed (storage plan, currently used bytes)
* relying on the `user.storage.usedBytes` (and updating it live on the UI), in the future we'll not even have to map it and send it to the browser
* the extra loop for contributors to show the storage quota of a different user (when all API calls are bound to room ID, the storage plan and quota in the response is always set correctly to that of the room owner, even if the API call was made by a contributor)

I will create a ticket to describe the necessary changes.

Reply:
Even though the paths are not quite completely RESTfull, I think your unification here is really good! 👏 And does eliminate quite some back and forth (in which risk of inconsistency of data can always creep in). Good work!

Thanks for the more extensive explanation in the JIRA ticket as well (clarifies point 4 from above for me, as I didn't see the code for it here).

One small suggestion I'd have, to ensure clarity, would be the same as for the case when the user deletes a private document in a room:
- let's call `handleDeleteRoomMedia` -> `handleDeleteOwnRoomMedia` (which also ties the `permissions.DELETE_OWN_PRIVATE_CONTENT` restriction into the name)
- let's call `storageService.deleteRoomMedia` -> `storageService.deleteOwnRoomMedia` (which also ties the inner `!isRoomOwner` condition into the name)

Reply:
Yeah, that adds more clarity. Renamed!